addons/stock/models/stock_move.py

```diff
@@ -357,6 +357,11 @@ def write(self, vals):
                 move_to_unreserve = self.filtered(lambda m: m.state not in ['draft', 'done', 'cancel'] and m.reserved_availability > vals.get('product_uom_qty'))
                 move_to_unreserve.do_unreserve()
                 (self - move_to_unreserve).filtered(lambda m: m.state == 'assigned').write({'state': 'partially_available'})
+        if 'move_line_ids' in vals and self.filtered(lambda m: m.state == 'done'):
```

Comment:
Same remark: who says that is you write on the move line it will do it through a move_line_ids on the move?